app/ui-react/packages/api/src/WithUserHelpers.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { ApiContext, IApiContext } from './ApiContext';
+import { callFetch } from './callFetch';
+
+export interface IWithUserHelpersChildrenProps {
+  logout(): Promise<any>;
+}
+export interface IWithUserHelpersProps {
+  children(props: IWithUserHelpersChildrenProps): any;
+}
+
+export class WithUserHelpersWrapped extends React.Component<
+  IWithUserHelpersProps & IApiContext
+> {
+  constructor(props: IWithUserHelpersProps & IApiContext) {
+    super(props);
+    this.logout = this.logout.bind(this);
+  }
+
+  public async logout() {
+    const response = await callFetch({
+      headers: {
+        ...this.props.headers,
+        responseType: 'arraybuffer',
+      },
+      method: 'GET',
+      url: `/logout`,
```

Comment:
Ok this is tricky. The logout page, while being a route, it's not _an app_ route, so we should handle it like an asset. I think this could help:

```tsx
url: `${process.env.PUBLIC_URL}/logout`, 
```

The thing is, on CI assets are not in the root folder, so we have to remember to always append the `PUBLIC_URL` bit to the url.